### PR TITLE
[1.17] crio wipe: version file persist

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
before, crio wipe wasn't precise enough. It always wiped both containers and images on a reboot and version bump. Unfortunately, if the version-file location was somewhere persistent, then no containers would be cleared on reboot.
    
We should always clear containers on reboot, but should really only clear images on an upgrade. As such, introduce a new flag: --version-file-persist. It should live somewhere persistent, so, if it doesn't exist (or crio is being upgraded, then we can be fairly certain an upgrade happened and it's worth upgrading.
    
Now, when version-file doesn't exist, but version-file-persist does, we only wipe the containers. If both exist and are the correct version, don't wipe anything. If neither exist, or version-file-persist is out of date, wipe them both
    
Also, update the crio wipe tests to test different scenerios and move the tests to a higher directory so they're run on PRs
#### Does this PR introduce a user-facing change?

```release-note
add --version-file-persist for finer grained control on crio wipe. version-file-persist should be in a persistent directory, and version-file should be in a temporary directory, if using crio wipe.
```

### CODE REVIEW

**Comment by haircommander** [comment]
one thing to note is upgrading to this version will mean, on first run of crio wipe, the images and containers will  be removed, despite it not being a minor version upgrade..

File: cmd/crio/wipe.go [hunk 1]
```diff
@@ -51,15 +51,15 @@ func crioWipe(c *cli.Context) error {
...
 	// if we should not wipe, exit with no error
-	// since we won't wipe images without wiping containers, we can just check the one condition
 	if !shouldWipeContainers {
+		// we should not wipe images without wiping containers
+		// in a future release, we should wipe both container and images if only shouldWipeImages is true.
+		// However, now, we cannot expect users to have version-file-persist after having upgraded
+		// to this version. Skip the wipe, for now, and log about it.
+		if shouldWipeImages {
+			logrus.Infof("legacy version-file path found, but new version-file-persist path not. Skipping wipe")
+		}
```
Comment (TomSweeneyRedHat) [inline cmd/crio/wipe.go:62]: Do we have a "TODO:" tag that we use to make sure we clean stuff up?  And/or is there an issue spun up for this that we could reference here?

Comment (haircommander) [reply cmd/crio/wipe.go:62]: I intend on cherry-picking this version into 1.18, then writing the new version in master. I am leaving myself in charge of remembering :smile:

**Pull request merged**